### PR TITLE
Improve downloading logs

### PR DESCRIPTION
**Pull request by pietroalbini**
This PR makes some improvements to the log downloading functionality of RLA, which I had to use to prepare the GHA training data. The two changes are:

* Urlencode the file names to avoid problems when job names include special chars.
* Run the log compression in the threadpool to avoid the download taking ages (compressing a single log took 30 seconds on my machine, and to train we need hundreds of them).

### CODE REVIEW

File: src/bin/offline/dl.rs [hunk 1]
```diff
@@ -91,7 +96,19 @@ pub fn download(
...
+            debug!("compression pool is busy, compressing on the main thread...");
+            offline::fs::save_compressed(&save_path, &data)?;
+        } else {
+            debug!("compressing on the pool...");
+            compression_pool.execute(move || {
+                crate::util::log_and_exit_error(move || {
+                    offline::fs::save_compressed(&save_path, &data)
+                });
+            });
+        }
```
Comment (Mark-Simulacrum) [inline src/bin/offline/dl.rs:111]: This seems fine, though a bit low-level. I guess rayon and a for_each method won't work? Have you tried that?

Comment (pietroalbini) [reply src/bin/offline/dl.rs:111]: Yep, I tried rayon, but it didn't like trait objects much.

Comment (Mark-Simulacrum) [reply src/bin/offline/dl.rs:111]: Huh. I would've expected `+ Send + Sync` on the trait object to be sufficient.

Comment (Mark-Simulacrum) [reply src/bin/offline/dl.rs:111]: (anyway don't treat this as blocking)

**Pull request merged**